### ISSUE TITLE
Wrong ordering of quotes

### ISSUE THREAD

**Issue by ragreener1** [jgm/citeproc#35]
This issue does not occur with pandoc-citeproc.

I am using pandoc 11.22 and citeproc 0.2. I am using chicago-author-date (the default) as the CSL

I am trying to cite an article with the title: 'On the Outside': Constructing Cycling Citizenship

With citeproc it gets rendered incorrectly as (double, single, double single):
“‘On the Outside”: Constructing Cycling Citizenship.’ Social & Cultural Geography 11 (1): 35–52. https://doi.org/10.1080/14649360903414593.

Whereas with pandoc-citeproc I get single, double, double, single:
Aldred, Rachel. 2010. ‘ “On the Outside”: Constructing Cycling Citizenship’. Social & Cultural Geography 11 (1): 35–52. https://doi.org/10.1080/14649360903414593.

**Comment by jgm** [comment]
Can you provide inputs and exact command to reproduce the issue?

**Comment by ragreener1** [comment]
Sure.

This is test.md
```
---
title: Test document
author: Test author
date: Test date
bibliography: library.json
---

This is a test citation [@Aldred2010].

# References

::: refs

:::
```

Here is library.json

```
[
	  {"id":"Aldred2010","accessed":{"date-parts":[[2020,11,16]]},"author":[{"family":"Aldred","given":"Rachel"}],"container-title":"Social & Cultural Geography","DOI":"10.1080/14649360903414593","ISSN":"1464-9365","issue":"1","issued":{"date-parts":[[2010,2,1]]},"note":"_eprint: https://doi.org/10.1080/14649360903414593","page":"35-52","publisher":"Routledge","source":"Taylor and Francis+NEJM","title":"'On the outside': constructing cycling citizenship","title-short":"‘On the outside’","type":"article-journal","URL":"https://doi.org/10.1080/14649360903414593","volume":"11"}
]
```

The command is `pandoc test.md -o test.pdf --citeproc --pdf-engine=lualatex`

Interestingly, without lualatex the behaviour is as expected. Though it works with `-F pandoc-citeproc` and lualatex as expected.
Also, if the references are in YAML in the frontmatter it also works as expected.

**Comment by jgm** [comment]
Useful to know it only affects lualatex.
Have you tried comparing the tex output?
```
pandoc test.md --citeproc -s -o test1.tex
pandoc test.md -F pandoc-citeproc -s -o test2.tex
diff test1.tex test2.tex
```

**Comment by ragreener1** [comment]
Here's the comparison:

```
< Aldred, Rachel. 2010. {``{`On the Outside'}: Constructing Cycling
< Citizenship.''} \emph{Social \& Cultural Geography} 11 (1): 35--52.
---
> Aldred, Rachel. 2010. ``\,`On the Outside': Constructing Cycling
> Citizenship.'' \emph{Social \& Cultural Geography} 11 (1): 35--52.
```

It appears that pandoc-citeproc is putting a thin space between the quotes.

**Comment by jgm** [comment]
Yes, that's a difference. But I don't see anything here that would create what you originally described: namely, double - single - double - single quote ordering.  Nor can I reproduce this when processing with lualatex.  Here's what I get with `pandoc -C -t latex 35.md -o 35a.pdf --pdf-engine=lualatex`.

<img width="557" alt="Screen Shot 2020-12-16 at 8 57 02 AM" src="https://user-images.githubusercontent.com/3044/102380465-ca739880-3f7c-11eb-8cf3-2cc62849b54b.png">

That looks fine to me.

**Comment by ragreener1** [comment]
After some more experimentation it seems that `lang: en-GB` must be set in the YAML metadata.
The error definitely occurs then. The diff is then:
```
< Aldred, Rachel. 2010. {`{``On the Outside''}: Constructing Cycling
< Citizenship.'} \emph{Social \& Cultural Geography} 11 (1): 35--52.
---
> Aldred, Rachel. 2010. `\,``On the Outside'': Constructing Cycling
> Citizenship'. \emph{Social \& Cultural Geography} 11 (1): 35--52.
```
Though, I'm wondering if this isn't an issue with citeproc rather something else with lualatex (perhaps babel given language makes a difference and the TeX source seems fine)...

![image](https://user-images.githubusercontent.com/2730998/102382481-68d61d00-3fc2-11eb-9dfd-5ab155130d6e.png)

**Comment by jgm** [comment]
Glad to get that pinned down.  I don't know why lualatex is behaving differently from other texs here.  I'm going to post something on tex stackexchange to see if I can find out.

**Comment by jgm** [comment]
Transferring to pandoc as it's not an issue with this library specifically.
